packages/react-magma-dom/src/components/Table/TablePagination.tsx

```diff
@@ -136,6 +136,55 @@ const RowsPerPageLabel = styled.span<{
   color: ${props => (props.isInverse ? props.theme.colors.neutral100 : '')};
 `;
 
+interface RowsPerPageControllerProps {
+  /**
+   * Event that fires when the number of rows per page changes
+   */
+  handleRowsPerPageChange?: (value: any) => void;
+  rowsPerPageValues: number[];
+  isInverse?: boolean;
+}
+
+const RowsPerPageController = ((props: RowsPerPageControllerProps) => {
+  const {
+    handleRowsPerPageChange,
+    rowsPerPageValues,
+    isInverse,
+  } = props;
+
+    const theme = React.useContext(ThemeContext);
+    const i18n = React.useContext(I18nContext);
+
+    const rowsPerPageItems = rowsPerPageValues.map(value => ({
+      label: value.toString(),
+      value,
+    }));
+
+  return (
+    <>
+      <RowsPerPageLabel isInverse={isInverse} theme={theme}>
+          {i18n.table.pagination.rowsPerPageLabel}:
+      </RowsPerPageLabel> 
+      <NativeSelect
+        onChange={(event) => handleRowsPerPageChange(event.target.value)}
+        aria-label={i18n.table.pagination.rowsPerPageLabel}
+        style={{minWidth: 80}}
+        testId="rowPerPageSelect" 
+        fieldId={''}
+      >
+        {rowsPerPageItems.map((row, index) => (
+          <option
+            key={index}
+            value={row.value}
+          >
+            {row.label}
+          </option>
+        ))}
+      </NativeSelect> 
```

Comment:
>  Since the event handler function for the two branches was different, the alternate implementation was non-functional.

Both controllers use the same `handleRowsPerPageChange` function, so I'd love to hear more about the issues you were seeing around this:
- https://github.com/cengage/react-magma/pull/1076/files#diff-72059da60baba78fdc1cdf4cd0725c54fb5c508fa988047877ad22f3e5c84eb8L252
- https://github.com/cengage/react-magma/pull/1076/files#diff-72059da60baba78fdc1cdf4cd0725c54fb5c508fa988047877ad22f3e5c84eb8L243

I would really prefer to keep both otherwise this is a breaking change, and we like to limit those as much as possible to make it easier for our adopters to stay up to date with the latest version.

Reply:
Perhaps I'm not understanding how the alternate codeflow is supposed to work, but if you go to the Table.stories.tsx file and make this change (on a branch without my changes) in either the UncontrolledPagination or ControlledPagination stories:
<img width="560" alt="Screenshot 2023-07-06 at 10 31 30 AM" src="https://github.com/cengage/react-magma/assets/88386737/584e2671-3469-42de-8ca6-ba9832d7e8e1">

Then you get a selector where you can make the selection, but it doesn't update at all:
![Screenshot 2023-07-06 at 10 31 12 AM](https://github.com/cengage/react-magma/assets/88386737/8a6b5f9e-a567-4da4-95b6-f52a77336aba)


Effectively, passing a value for dropdownDropDirection completely breaks rows per page selection. 
DropdownMenuItem's onClick function has this signature: `onClick?: (value?: string | number) => void;`
NativeSelect's onChange function has this signature:` onChange?: (event: ChangeEvent) => void;`


handleRowsPerPageChange is written to accept an `event` parameter, not a `value` parameter, so when DropdownMenuItem calls it, it breaks and fails to call setRowsPerPageState or onRowsPerPageChange appropriately: 

<img width="387" alt="Screenshot 2023-07-06 at 10 56 54 AM" src="https://github.com/cengage/react-magma/assets/88386737/15eac90b-1e66-4678-929d-642cb1deb359">

![Screenshot 2023-07-06 at 10 55 35 AM](https://github.com/cengage/react-magma/assets/88386737/016128a6-02d6-47e5-b458-f26fd46cb7e9)

So this is a bug that no one has complained about because on one is passing the dropdownDropDirection parameter.

Reply:
I see what you're saying. In that case, I agree with you. Lets get rid of `dropdownDropDirection` and we'll call this a `minor` change-- it slightly breaks the rules, but like you said, no one has complained in months about this and using NativeSelect is ultimately the best solution. Thanks for digging into this!